main.cpp

```diff
@@ -8,17 +8,25 @@
 
 DigitalOut led1(LED1);
 
+#define SLEEP_TIME                  500 // (msec)
+#define PRINT_AFTER_N_LOOPS         20
+
 // main() runs in its own thread in the OS
 int main()
 {
-    SystemReport sys_state(500 /* Loop delay time in ms */);
+    SystemReport sys_state( SLEEP_TIME * PRINT_AFTER_N_LOOPS /* Loop delay time in ms */);
 
+    int count = 0;
     while (true) {
         // Blink LED and wait 0.5 seconds
         led1 = !led1;
-        wait(0.5f);
+        wait_ms(SLEEP_TIME);
 
-        // Following the main thread wait, report on the current system status
-        sys_state.report_state();
+        if ((0 == count) || (PRINT_AFTER_N_LOOPS == count)) {
```

Comment:
```suggestion
        if ((count == 0) || (count == PRINT_AFTER_N_LOOPS)) {
```

Reply:
Will stick to constants on left side of comparison :-)